src/conduit/openapi.py

```diff
@@ -22,6 +25,23 @@ def includeme(config: Configurator) -> None:
     config.pyramid_openapi3_add_explorer(route="/api")
 
 
+def object_or_404(obj: t.Optional[t.Any]) -> t.Any:
+    """Use this to stop view code execution if object is not found.
+
+    If obj is None, return response with 404 status code.
+    Otherwise, continue with view code.
+
+    Example usage:
+
+    user = object_or_404(User.by_username("foo", db=request.db))
+    ... # code here knows that user exists and does not need to do any checks
+    """
+    if obj is None:
+        raise HTTPNotFound
+    else:
+        return obj
+
+
 @exception_view_config(Exception)
 def unknown_error(exc: Exception, request: Request):
```

Comment:
Return type `-> Exception`: All classes.

Reply:
It's actually HTTPUnprocessableEntity, but still thanks for the reminder that I have not declared a return type.